### PR TITLE
support multiple "distros"

### PR DESCRIPTION
**Pull request by esell**
This is in regards to #6. It'll allow users to create various "distros" inside of the apt repo.

### CODE REVIEW

**Comment by esell** [comment]
@icholy, as the only other contributor I'd be interested to see what you think...

File: main.go [hunk 1]
```diff
@@ -27,18 +27,20 @@ type Conf struct {
...
+	DistroNames  []string `json:"distroNames"`
 	EnableSSL    bool     `json:"enableSSL"`
 	SSLCert      string   `json:"SSLcert"`
 	SSLKey       string   `json:"SSLkey"`
 }
 
-func (c Conf) ArchPath(arch string) string {
-	return filepath.Join(c.RootRepoPath, "/dists/stable/main/binary-"+arch)
+func (c Conf) ArchPath(distro, arch string) string {
+	return filepath.Join(c.RootRepoPath, "/dists/"+distro+"/main/binary-"+arch)
```
Comment (icholy) [inline main.go]:
``` go
return filepath.Join(c.RootRepoPath, "dists", distro, "main/binary-"+arch)
```

File: main.go [hunk 2]
```diff
@@ -230,9 +234,13 @@ func uploadHandler(config Conf) http.Handler {
 			return
 		}
 		archType := r.URL.Query().Get("arch")
+		distroName := r.URL.Query().Get("distro")
```
Comment (icholy) [inline main.go]: I'd move this line below. Above the default value assignment.

File: main.go [hunk 3]
```diff
@@ -76,15 +78,17 @@ func main() {
...
-				log.Printf("Directory for %s does not exist, creating", arch)
-				if err := os.MkdirAll(config.ArchPath(arch), 0755); err != nil {
-					return fmt.Errorf("error creating directory for %s: %s", arch, err)
+	for _, distro := range config.DistroNames {
+		for _, arch := range config.SupportArch {
+			if _, err := os.Stat(config.ArchPath(distro, arch)); err != nil {
+				if os.IsNotExist(err) {
+					log.Printf("Directory for %s does not exist, creating", arch)
+					if err := os.MkdirAll(config.ArchPath(distro, arch), 0755); err != nil {
+						return fmt.Errorf("error creating directory for %s: %s", arch, err)
```
Comment (icholy) [inline main.go]: These messages should include the distro. Something like:

`return fmt.Errorf("error creating directory for %s (%s): %s", distro, arch, err)`

`log.Printf("Directory for %s (%s) does not exist, creating", distro, arch)`

File: sample_conf.json [hunk 4]
```diff
@@ -2,6 +2,7 @@
 	"listenPort" : "9090",
 	"rootRepoPath" : "/opt/repo",
 	"supportedArch" : ["all","i386","amd64"],
+        "distroNames":["stable"],
```
Comment (icholy) [inline sample_conf.json]: indentation

**Comment by icholy** [comment]
@esell added some suggestions.

**Comment by esell** [comment]
cool, thanks for the feedback. had some time to kill in the airport so i figured i'd have a go at knocking this out so now that it's cleaned up a bit i'll merge it in.

**Pull request merged**